lib/src/core/services/kana_service.dart

```diff
@@ -0,0 +1,72 @@
+import 'dart:convert';
+
+import 'package:isar/isar.dart';
+import 'package:kana_to_kanji/src/core/models/kana.dart';
+import 'package:kana_to_kanji/src/core/services/api_service.dart';
+import 'package:kana_to_kanji/src/locator.dart';
+import 'package:http/http.dart' as http;
+
+class KanaService {
+  final Isar _isar = locator<Isar>();
+  final ApiService _apiService = locator<ApiService>();
+
+  /// Get all the kana related to the group ids given in parameter.
+  Future<List<Kana>> getByGroupIds(List<int> groupIds) async {
+    if (groupIds.isEmpty) {
+      return Future.value(_isar.kanas.where().findAll());
+    }
+
+    var kanaQuery = _isar.kanas.where().groupIdEqualTo(groupIds[0]);
+
+    for (var i = 1; i < groupIds.length; i++) {
+      kanaQuery = kanaQuery.or().groupIdEqualTo(groupIds[i]);
+    }
+
+    final kanas = kanaQuery.findAll();
+
+    if (kanas.isEmpty) {
```

Comment:
If we don't find any kana from the groups specified. We delete all the kana data, reload everything and re-do the search? I don't understand the reason 😓

Reply:
Vu qu'on les charge d'un coup, on n'est pas censé avoir de groupes vides. Du coup, on recharge tout.